ompi/mca/coll/sync/coll_sync_module.c

```diff
@@ -45,18 +45,34 @@ static void mca_coll_sync_module_construct(mca_coll_sync_module_t *module)
     memset(&(module->c_coll), 0, sizeof(module->c_coll));
     module->before_num_operations = 0;
     module->after_num_operations = 0;
+    module->before_num_operations_alltoallv = 0;
+    module->after_num_operations_alltoallv = 0;
+    module->before_num_operations_alltoallw = 0;
+    module->after_num_operations_alltoallw = 0;
     module->in_operation = false;
 }
 
 static void mca_coll_sync_module_destruct(mca_coll_sync_module_t *module)
 {
-    OBJ_RELEASE(module->c_coll.coll_bcast_module);
-    OBJ_RELEASE(module->c_coll.coll_gather_module);
-    OBJ_RELEASE(module->c_coll.coll_gatherv_module);
-    OBJ_RELEASE(module->c_coll.coll_reduce_module);
-    OBJ_RELEASE(module->c_coll.coll_reduce_scatter_module);
-    OBJ_RELEASE(module->c_coll.coll_scatter_module);
-    OBJ_RELEASE(module->c_coll.coll_scatterv_module);
+    if (0 != mca_coll_sync_component.barrier_before_nops ||
+        0 != mca_coll_sync_component.barrier_after_nops) {
+        OBJ_RELEASE(module->c_coll.coll_alltoallv_module);
+        OBJ_RELEASE(module->c_coll.coll_alltoallw_module);
+        OBJ_RELEASE(module->c_coll.coll_bcast_module);
+        OBJ_RELEASE(module->c_coll.coll_gather_module);
+        OBJ_RELEASE(module->c_coll.coll_gatherv_module);
+        OBJ_RELEASE(module->c_coll.coll_reduce_module);
+        OBJ_RELEASE(module->c_coll.coll_reduce_scatter_module);
+        OBJ_RELEASE(module->c_coll.coll_scatter_module);
+        OBJ_RELEASE(module->c_coll.coll_scatterv_module);
+    } else {
```

Comment:
I don't think this should be an 'else' just a close bracket.  Couldn't someone run with both?  Of course we don't want to obj_release twice, so might need to rework the logic a bit.